### ISSUE TITLE
Applications failed to start successfully reported but juju status shows all units ready

### ISSUE THREAD

**Issue by jogeo** [conjure-up/conjure-up#1218]
With conjure-up 2.4.0 and MAAS version: 2.2.2 (6099-g8751f91-0ubuntu1~16.04.1) and juju 2.2.6-xenial-amd64:

```
$ conjure-up --nosync --spells-dir /home/cdoqa/spells.modified/ openstack-base oil-staging conjure default
[info] Summoning openstack-base to oil-staging
[info] Creating Juju model.
[info] Juju model created.
[info] Adding machines: 0: (xenial, ), 1: (xenial, ), 2: (xenial, ), 3: (xenial, )
[info] Added machines: 0, 1, 2, 3
[info] Deploying mysql...
[info] Deploying neutron-api...
[info] Deploying neutron-gateway...
[info] Deploying neutron-openvswitch...
[info] Deploying nova-cloud-controller...
[info] Deploying nova-compute...
[info] Deploying ceph-osd...
[info] Deploying ntp...
[info] Deploying ceph-radosgw...
[info] Deploying openstack-dashboard...
[info] Deploying cinder...
[info] Deploying rabbitmq-server...
[info] Deploying cinder-ceph...
[info] Deploying glance...
[info] Deploying ceph-mon...
[info] Deploying keystone...
[info] neutron-openvswitch: deployed, installing.
[info] neutron-api: deployed, installing.
[info] cinder-ceph: deployed, installing.
[info] mysql: deployed, installing.
[info] glance: deployed, installing.
[info] keystone: deployed, installing.
[info] rabbitmq-server: deployed, installing.
[info] ceph-mon: deployed, installing.
[info] ceph-radosgw: deployed, installing.
[info] neutron-gateway: deployed, installing.
[info] Setting relation ceph-mon:radosgw <-> ceph-radosgw:mon
[info] ceph-osd: deployed, installing.
[info] Setting relation ceph-mon:osd <-> ceph-osd:mon
[info] openstack-dashboard: deployed, installing.
[info] Setting relation keystone:identity-service <-> openstack-dashboard:identity-service
[info] nova-compute: deployed, installing.
[info] Setting relation neutron-openvswitch:neutron-plugin <-> nova-compute:neutron-plugin
[info] Setting relation ceph-radosgw:identity-service <-> keystone:identity-service
[info] Setting relation ceph-mon:client <-> cinder-ceph:ceph
[info] nova-cloud-controller: deployed, installing.
[info] cinder: deployed, installing.
[info] ntp: deployed, installing.
[info] Setting relation neutron-api:neutron-plugin-api <-> neutron-gateway:neutron-plugin-api
[info] Setting relation mysql:shared-db <-> neutron-api:shared-db
[info] Setting relation neutron-gateway:amqp <-> rabbitmq-server:amqp
[info] Setting relation neutron-gateway:juju-info <-> ntp:juju-info
[info] Setting relation cinder:amqp <-> rabbitmq-server:amqp
[info] Setting relation cinder-ceph:ceph-access <-> nova-compute:ceph-access
[info] Setting relation cinder:image-service <-> glance:image-service
[info] Setting relation glance:image-service <-> nova-compute:image-service
[info] Setting relation keystone:identity-service <-> neutron-api:identity-service
[info] Setting relation cinder:cinder-volume-service <-> nova-cloud-controller:cinder-volume-service
[info] Setting relation cinder-ceph:storage-backend <-> cinder:storage-backend
[info] Setting relation neutron-openvswitch:amqp <-> rabbitmq-server:amqp
[info] Setting relation ceph-mon:client <-> glance:ceph
[info] Setting relation neutron-api:neutron-api <-> nova-cloud-controller:neutron-api
[info] Setting relation mysql:shared-db <-> nova-cloud-controller:shared-db
[info] Setting relation neutron-gateway:quantum-network-service <-> nova-cloud-controller:quantum-network-service
[info] Setting relation nova-compute:amqp <-> rabbitmq-server:amqp
[info] Setting relation glance:amqp <-> rabbitmq-server:amqp
[info] Setting relation keystone:identity-service <-> nova-cloud-controller:identity-service
[info] Setting relation glance:image-service <-> nova-cloud-controller:image-service
[info] Setting relation nova-compute:juju-info <-> ntp:juju-info
[info] Setting relation nova-cloud-controller:amqp <-> rabbitmq-server:amqp
[info] Setting relation cinder:identity-service <-> keystone:identity-service
[info] Setting relation neutron-api:neutron-plugin-api <-> neutron-openvswitch:neutron-plugin-api
[info] Setting relation ceph-mon:client <-> nova-compute:ceph
[info] Setting relation nova-cloud-controller:cloud-compute <-> nova-compute:cloud-compute
[info] Setting relation glance:shared-db <-> mysql:shared-db
[info] Setting relation neutron-api:amqp <-> rabbitmq-server:amqp
[info] Setting relation keystone:shared-db <-> mysql:shared-db
[info] Setting relation glance:identity-service <-> keystone:identity-service
[info] Setting relation cinder:shared-db <-> mysql:shared-db
[info] Waiting for deployment to settle.
[error] Some applications failed to start successfully.
[warning] Shutting down

$ juju status
Model    Controller  Cloud/Region  Version      SLA
default  conjure     oil-staging   2.3-beta2.1  unsupported

App                    Version       Status  Scale  Charm                  Store       Rev  OS      Notes
ceph-mon               12.2.0        active      3  ceph-mon               jujucharms   12  ubuntu  
ceph-osd               12.2.0        active      3  ceph-osd               jujucharms  246  ubuntu  
ceph-radosgw           12.2.0        active      1  ceph-radosgw           jujucharms  251  ubuntu  
cinder                 11.0.0        active      1  cinder                 jujucharms  264  ubuntu  
cinder-ceph            11.0.0        active      1  cinder-ceph            jujucharms  227  ubuntu  
glance                 15.0.0        active      1  glance                 jujucharms  259  ubuntu  
keystone               12.0.0        active      1  keystone               jujucharms  268  ubuntu  
mysql                  5.6.34-26.19  active      1  percona-cluster        jujucharms  254  ubuntu  
neutron-api            11.0.1        active      1  neutron-api            jujucharms  252  ubuntu  
neutron-gateway        11.0.1        active      1  neutron-gateway        jujucharms  238  ubuntu  
neutron-openvswitch    11.0.1        active      3  neutron-openvswitch    jujucharms  244  ubuntu  
nova-cloud-controller  16.0.1        active      1  nova-cloud-controller  jujucharms  300  ubuntu  
nova-compute           16.0.1        active      3  nova-compute           jujucharms  273  ubuntu  
ntp                    4.2.8p4+dfsg  active      4  ntp                    jujucharms   18  ubuntu  
openstack-dashboard    12.0.0        active      1  openstack-dashboard    jujucharms  250  ubuntu  
rabbitmq-server        3.5.7         active      1  rabbitmq-server        jujucharms   65  ubuntu  

Unit                      Workload  Agent      Machine  Public address  Ports              Message
ceph-mon/0                active    idle       0/lxd/1  10.244.241.79                      Unit is ready and clustered
ceph-mon/1                active    idle       1/lxd/0  10.244.241.183                     Unit is ready and clustered
ceph-mon/2*               active    idle       2/lxd/1  10.244.241.73                      Unit is ready and clustered
ceph-osd/0                active    idle       0        10.244.241.18                      Unit is ready (1 OSD)
ceph-osd/1                active    idle       1        10.244.241.54                      Unit is ready (1 OSD)
ceph-osd/2*               active    idle       2        10.244.204.219                     Unit is ready (1 OSD)
ceph-radosgw/0*           active    executing  3/lxd/2  10.244.241.40   80/tcp             (update-status) Unit is ready
cinder/0*                 active    idle       0/lxd/2  10.244.241.76   8776/tcp           Unit is ready
  cinder-ceph/0*          active    idle                10.244.241.76                      Unit is ready
glance/0*                 active    idle       1/lxd/1  10.244.241.47   9292/tcp           Unit is ready
keystone/0*               active    idle       2/lxd/0  10.244.205.9    5000/tcp           Unit is ready
mysql/0*                  active    idle       3/lxd/0  10.244.204.208  3306/tcp           Unit is ready
neutron-api/0*            active    idle       0/lxd/0  10.244.243.0    9696/tcp           Unit is ready
neutron-gateway/0*        active    idle       3        10.244.241.31                      Unit is ready
  ntp/0*                  active    idle                10.244.241.31   123/udp            Unit is ready
nova-cloud-controller/0*  active    idle       1/lxd/2  10.244.241.59   8774/tcp,8778/tcp  Unit is ready
nova-compute/0            active    idle       0        10.244.241.18                      Unit is ready
  neutron-openvswitch/2   active    idle                10.244.241.18                      Unit is ready
  ntp/3                   active    idle                10.244.241.18   123/udp            Unit is ready
nova-compute/1            active    idle       1        10.244.241.54                      Unit is ready
  neutron-openvswitch/1   active    idle                10.244.241.54                      Unit is ready
  ntp/2                   active    executing           10.244.241.54   123/udp            (update-status) Unit is ready
nova-compute/2*           active    idle       2        10.244.204.219                     Unit is ready
  neutron-openvswitch/0*  active    idle                10.244.204.219                     Unit is ready
  ntp/1                   active    idle                10.244.204.219  123/udp            Unit is ready
openstack-dashboard/0*    active    idle       2/lxd/2  10.244.241.181  80/tcp,443/tcp     Unit is ready
rabbitmq-server/0*        active    idle       3/lxd/1  10.244.241.8    5672/tcp           Unit is ready

Machine  State    DNS             Inst id              Series  AZ       Message
0        started  10.244.241.18   4y4crm               xenial  default  Deployed
0/lxd/0  started  10.244.243.0    juju-7c4978-0-lxd-0  xenial  default  Container started
0/lxd/1  started  10.244.241.79   juju-7c4978-0-lxd-1  xenial  default  Container started
0/lxd/2  started  10.244.241.76   juju-7c4978-0-lxd-2  xenial  default  Container started
1        started  10.244.241.54   4y4crp               xenial  default  Deployed
1/lxd/0  started  10.244.241.183  juju-7c4978-1-lxd-0  xenial  default  Container started
1/lxd/1  started  10.244.241.47   juju-7c4978-1-lxd-1  xenial  default  Container started
1/lxd/2  started  10.244.241.59   juju-7c4978-1-lxd-2  xenial  default  Container started
2        started  10.244.204.219  4y3hdy               xenial  default  Deployed
2/lxd/0  started  10.244.205.9    juju-7c4978-2-lxd-0  xenial  default  Container started
2/lxd/1  started  10.244.241.73   juju-7c4978-2-lxd-1  xenial  default  Container started
2/lxd/2  started  10.244.241.181  juju-7c4978-2-lxd-2  xenial  default  Container started
3        started  10.244.241.31   4y3he4               xenial  default  Deployed
3/lxd/0  started  10.244.204.208  juju-7c4978-3-lxd-0  xenial  default  Container started
3/lxd/1  started  10.244.241.8    juju-7c4978-3-lxd-1  xenial  default  Container started
3/lxd/2  started  10.244.241.40   juju-7c4978-3-lxd-2  xenial  default  Container started

Relation provider                        Requirer                                       Interface               Type
ceph-mon:client                          cinder-ceph:ceph                               ceph-client             regular
ceph-mon:client                          glance:ceph                                    ceph-client             regular
ceph-mon:client                          nova-compute:ceph                              ceph-client             regular
ceph-mon:mon                             ceph-mon:mon                                   ceph                    peer
ceph-mon:osd                             ceph-osd:mon                                   ceph-osd                regular
ceph-mon:radosgw                         ceph-radosgw:mon                               ceph-radosgw            regular
ceph-radosgw:cluster                     ceph-radosgw:cluster                           swift-ha                peer
cinder-ceph:ceph-access                  nova-compute:ceph-access                       cinder-ceph-key         regular
cinder-ceph:storage-backend              cinder:storage-backend                         cinder-backend          subordinate
cinder:cinder-volume-service             nova-cloud-controller:cinder-volume-service    cinder                  regular
cinder:cluster                           cinder:cluster                                 cinder-ha               peer
glance:cluster                           glance:cluster                                 glance-ha               peer
glance:image-service                     cinder:image-service                           glance                  regular
glance:image-service                     nova-cloud-controller:image-service            glance                  regular
glance:image-service                     nova-compute:image-service                     glance                  regular
keystone:cluster                         keystone:cluster                               keystone-ha             peer
keystone:identity-service                ceph-radosgw:identity-service                  keystone                regular
keystone:identity-service                cinder:identity-service                        keystone                regular
keystone:identity-service                glance:identity-service                        keystone                regular
keystone:identity-service                neutron-api:identity-service                   keystone                regular
keystone:identity-service                nova-cloud-controller:identity-service         keystone                regular
keystone:identity-service                openstack-dashboard:identity-service           keystone                regular
mysql:cluster                            mysql:cluster                                  percona-cluster         peer
mysql:shared-db                          cinder:shared-db                               mysql-shared            regular
mysql:shared-db                          glance:shared-db                               mysql-shared            regular
mysql:shared-db                          keystone:shared-db                             mysql-shared            regular
mysql:shared-db                          neutron-api:shared-db                          mysql-shared            regular
mysql:shared-db                          nova-cloud-controller:shared-db                mysql-shared            regular
neutron-api:cluster                      neutron-api:cluster                            neutron-api-ha          peer
neutron-api:neutron-api                  nova-cloud-controller:neutron-api              neutron-api             regular
neutron-api:neutron-plugin-api           neutron-gateway:neutron-plugin-api             neutron-plugin-api      regular
neutron-api:neutron-plugin-api           neutron-openvswitch:neutron-plugin-api         neutron-plugin-api      regular
neutron-gateway:cluster                  neutron-gateway:cluster                        quantum-gateway-ha      peer
neutron-gateway:juju-info                ntp:juju-info                                  juju-info               subordinate
neutron-gateway:quantum-network-service  nova-cloud-controller:quantum-network-service  quantum                 regular
neutron-openvswitch:neutron-plugin       nova-compute:neutron-plugin                    neutron-plugin          subordinate
nova-cloud-controller:cluster            nova-cloud-controller:cluster                  nova-ha                 peer
nova-compute:cloud-compute               nova-cloud-controller:cloud-compute            nova-compute            regular
nova-compute:compute-peer                nova-compute:compute-peer                      nova                    peer
nova-compute:juju-info                   ntp:juju-info                                  juju-info               subordinate
ntp:ntp-peers                            ntp:ntp-peers                                  ntp                     peer
openstack-dashboard:cluster              openstack-dashboard:cluster                    openstack-dashboard-ha  peer
rabbitmq-server:amqp                     cinder:amqp                                    rabbitmq                regular
rabbitmq-server:amqp                     glance:amqp                                    rabbitmq                regular
rabbitmq-server:amqp                     neutron-api:amqp                               rabbitmq                regular
rabbitmq-server:amqp                     neutron-gateway:amqp                           rabbitmq                regular
rabbitmq-server:amqp                     neutron-openvswitch:amqp                       rabbitmq                regular
rabbitmq-server:amqp                     nova-cloud-controller:amqp                     rabbitmq                regular
rabbitmq-server:amqp                     nova-compute:amqp                              rabbitmq                regular
rabbitmq-server:cluster                  rabbitmq-server:cluster                        rabbitmq-ha             peer
```

**Comment by jogeo** [comment]
[conjure-up_logs.tar.gz](https://github.com/conjure-up/conjure-up/files/1451384/conjure-up_logs.tar.gz)

**Comment by johnsca** [comment]
The `deploy-wait.err` file is there, but it's empty.  It does seem to be waiting for just over an hour before failing, so I would expect some output in there.

**Comment by jogeo** [comment]
cdoqa@conjurejog:~$ juju wait -vw
INFO:root:All units idle since 2017-11-07 21:46:31.086032Z (ceph-mon/0, ceph-mon/1, ceph-mon/2, ceph-osd/0, ceph-osd/1, ceph-osd/2, ceph-radosgw/0, cinder/0, glance/0, keystone/0, mysql/0, neutron-api/0, neutron-gateway/0, nova-cloud-controller/0, nova-compute/0, nova-compute/1, nova-compute/2, openstack-dashboard/0, rabbitmq-server/0)
DEBUG:root:ceph-mon is lead by ceph-mon/0
DEBUG:root:ceph-osd is lead by ceph-osd/0
DEBUG:root:ceph-radosgw is lead by ceph-radosgw/0
DEBUG:root:cinder is lead by cinder/0
DEBUG:root:cinder-ceph is lead by cinder-ceph/0
DEBUG:root:glance is lead by glance/0
DEBUG:root:keystone is lead by keystone/0
DEBUG:root:mysql is lead by mysql/0
DEBUG:root:neutron-api is lead by neutron-api/0
DEBUG:root:neutron-gateway is lead by neutron-gateway/0
DEBUG:root:ntp is lead by ntp/0
DEBUG:root:nova-cloud-controller is lead by nova-cloud-controller/0
DEBUG:root:nova-compute is lead by nova-compute/0
DEBUG:root:neutron-openvswitch is lead by neutron-openvswitch/0
DEBUG:root:openstack-dashboard is lead by openstack-dashboard/0
DEBUG:root:rabbitmq-server is lead by rabbitmq-server/0

**Comment by battlemidget** [comment]
This should be fixed with #1242 can you try and reproduce with --edge?
